syntax = "proto3";

package writer;

import "google/protobuf/timestamp.proto";

service WriterService {
    rpc CreateAlbum           (CreateAlbum_Request)           returns (CreateAlbum_Response);
    rpc RemoveAlbum           (RemoveAlbum_Request)           returns (RemoveAlbum_Response);
    rpc RemoveAlbumsByOwnerID (RemoveAlbumsByOwnerID_Request) returns (RemoveAlbumsByOwnerID_Response);

    rpc CreateTrack           (CreateTrack_Request)           returns (CreateTrack_Response);
    rpc RemoveTrack           (RemoveTrack_Request)           returns (RemoveTrack_Response);
}

message CreateAlbum_Request {
    string title = 1;
    int32 owner_id = 2;
    google.protobuf.Timestamp release_date = 3;
}

message CreateAlbum_Response {
    int32 album_id = 1;
}

message RemoveAlbum_Request {
    int32 album_id = 1;
}

message RemoveAlbum_Response {
    int32 album_id = 1;
}

message RemoveAlbumsByOwnerID_Request {
    int32 owner_id = 1;
}

message RemoveAlbumsByOwnerID_Response {
    repeated int32 album_ids = 1;
}

message CreateTrack_Request {
    string title = 1;
    int32 album_id = 2;
    bool explicit = 3;
    repeated string genre_names = 4;
}

message CreateTrack_Response {
    int32 track_id = 1;
}

message RemoveTrack_Request {
    int32 track_id = 1;
}

message RemoveTrack_Response {
    int32 track_id = 1;
}